contracts/EventToken.sol

```diff
@@ -6,39 +6,65 @@ contract EventToken {
     ERC20 erc20Contract;
     uint256 supplyLimit;
     uint256 currentSupply;
+    uint256 basePriceOfToken;
     address owner;
 
-    event GetEventToken(address to, uint256 amount);
-    event RefundEventToken(address to, uint256 amount);
+    event MintToken(address to, uint256 amount);
+    event BurnToken(address to, uint256 amount);
 
     constructor() public {
         ERC20 e = new ERC20();
         erc20Contract = e;
         owner = msg.sender;
         currentSupply = 0;
+        basePriceOfToken = 50000;
     }
 
     modifier onlyOwner() {
-        require(msg.sender == owner);
+        require(msg.sender == owner, "You do not have permission to do this");
         _;
     }
 
-    // For purchasing of Tokens
-    function getEventToken() public payable {
-        uint256 amt = msg.value / 50000;
-        // require(erc20Contract.totalSupply() + amt < supplyLimit, "CT supply is not enough");
-        erc20Contract.mint(msg.sender, amt);
-        currentSupply = currentSupply + amt;
-        emit GetEventToken(msg.sender, amt);
+    /**
+     * Mint event tokens for address based on amount of wei
+     *
+     * param amount of token
+     * param address to get token
+     * 
+     */
+    function mintToken(uint256 amtOfWei, address _to) public onlyOwner {
+        uint256 amtOfToken = amtOfWei / basePriceOfToken;
```

Comment:
"Mint token based on Eth given in input. The eth amount should be 5% of price of ticket."
Are the tokens given to buyers supposed to be 5% of the ticket price? The code here seems to be minting tokens based on 1 usd (50000 wei) : 1 token

Reply:
5% of ticket price in wei, converted to tokens which we will price it as per token is 1 USD as the base price. We can discuss on the base price accordingly if 1 USD is deem too high/low